### PR TITLE
[PAY-1464] DMs: Go to chat after successful tip

### PR DESCRIPTION
**Pull request by rickyrombo**
### Description

Notably, this does not give us a nice pretty UI experience explaining what's going on. It takes the user to the chat page (or chat screen) and shows them a spinner on it until the tip is confirmed and the chat can be created. This takes around 20 seconds but starts immediately after success, whereas the user won't get navigated until they close the tip modal or screen.

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

Tested on iOS simulator e2e with stage data, as well as e2e desktop web pointed at stage

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/common/src/store/pages/chat/sagas.ts [hunk 1]
```diff
@@ -18,7 +18,11 @@ import { ulid } from 'ulid'
 import { ID } from 'models/Identifiers'
 import { Status } from 'models/Status'
 import { getAccountUser, getUserId } from 'store/account/selectors'
+<<<<<<< HEAD
```
Comment (dharit-tan) [inline packages/common/src/store/pages/chat/sagas.ts:21]: conflict markers

File: packages/web/src/pages/chat-page/components/CreateChatUserResult.tsx [hunk 2]
```diff
@@ -109,6 +109,7 @@ export const MessageUserSearchResult = (props: UserResultComposeProps) => {
     if (canCreateChat) {
       closeParentModal()
       dispatch(createChat({ userIds: [user.user_id] }))
+      closeParentModal()
```
Comment (dharit-tan) [inline packages/web/src/pages/chat-page/components/CreateChatUserResult.tsx:112]: this seems odd. Do we need this bc there are 2 layers of parent modals?

File: packages/web/src/components/tipping/tip-audio/TipAudioModal.tsx [hunk 3]
```diff
@@ -111,14 +114,31 @@ export const TipAudioModal = () => {
...
+  const currentUserId = useSelector(accountSelectors.getUserId)
 
   const audioFeaturesDegradedText = useRemoteVar(
     StringKeys.AUDIO_FEATURES_DEGRADED_TEXT
   )
 
   const onClose = useCallback(() => {
+    // After success + close, take the user to the chat they were
+    // attempting to make if they were unlocking DMs by tipping.
+    // The saga will create the chat once the tip is confrimed
```
Comment (dharit-tan) [inline packages/web/src/components/tipping/tip-audio/TipAudioModal.tsx:127]: typo - confirmed

File: packages/web/src/common/store/tipping/sagas.ts [hunk 4]
```diff
@@ -376,12 +376,23 @@ function* sendTipAsync() {
...
       yield* put(
         fetchPermissions({ userIds: [sender.user_id, recipient.user_id] })
       )
+      if (source === 'inboxUnavailableModal') {
+        console.debug('Creating chat silently...')
+        // Create the chat but don't navigate
+        yield* put(
+          chatActions.createChat({
+            userIds: [recipient.user_id],
+            skipNavigation: true
```
Comment (dharit-tan) [inline packages/web/src/common/store/tipping/sagas.ts:392]: Just curious - why do we need to skip navigation? If we're already in the right chat (with loading spinner) and we navigate to it _again_, does it look bad?

**Review by dharit-tan — APPROVED** [review]

File: packages/web/src/common/store/tipping/sagas.ts (hunk 4 above)
Comment (rickyrombo) [reply packages/web/src/common/store/tipping/sagas.ts:392]: The user's still looking at the tip sent screen. We don't want to navigate unless they close it

File: packages/web/src/pages/chat-page/components/CreateChatUserResult.tsx (hunk 2 above)
Comment (rickyrombo) [reply packages/web/src/pages/chat-page/components/CreateChatUserResult.tsx:112]: nah, bad merge, 110 takes care of this

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-dms-tip-chat

**Pull request merged**